cmd/syncthing/main.go

```diff
@@ -719,9 +709,51 @@ func syncthingMain() {
 		mainSvc.Add(relaySvc)
 	}
 
+	// Start discovery
+
+	cachedDiscovery := discover.NewCachingMux()
+	mainSvc.Add(cachedDiscovery)
+
+	if cfg.Options().GlobalAnnEnabled {
+		insecure := cfg.Options().GlobalAnnInsecure
```

Comment:
Then again... For the relay stuff this is one thing, because we extract basically the host:port from the URL and we know the rest is parameters for us. For announce, we're actually posting to the given HTTPS URL. So we get `https://discover.example.com/?insecure=true`. Are we supposed to post to that URL (maybe the insecure flag means something to the server), or eat the query part and assume it was meant for us to ignore the cert?